### PR TITLE
DT-2122: ✨ Add in gp mappings

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: src/main/resources/application-localstack.yml [hunk 1]
```diff
@@ -2,6 +2,3 @@ server.shutdown: immediate
 
 oauth:
   endpoint.url: http://localhost:8090/auth
-
-aws:
```
Comment (petergphillips) [inline src/main/resources/application-localstack.yml:6]: not used

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonregister/model/PrisonServiceTest.kt [hunk 2]
```diff
@@ -36,59 +37,71 @@ class PrisonServiceTest {
...
+      whenever(prisonRepository.findByGpPracticeGpPracticeCode(anyString())).thenReturn(prison)
+      val prisonDto = prisonService.findByGpPractice("MDI")
+      assertThat(prisonDto).isEqualTo(prison)
+    }
+  }
+
   @Nested
   inner class VCC {
-    private val PRISON_ID = "MDI"
+    private val prisonId = "MDI"
```
Comment (petergphillips) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonregister/model/PrisonServiceTest.kt:54]: fix to intellij warning about private variable being in capitals

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonregister/model/PrisonService.kt [hunk 3]
```diff
@@ -10,27 +10,25 @@ import javax.persistence.EntityNotFoundException
...
-    val prison =
-      prisonRepository.findByIdOrNull(prisonId) ?: throw EntityNotFoundException("Prison $prisonId not found")
-    return PrisonDto(prison)
-  }
+  fun findById(prisonId: String): Prison =
+    prisonRepository.findByIdOrNull(prisonId) ?: throw EntityNotFoundException("Prison $prisonId not found")
+
+  fun findByGpPractice(gpPracticeCode: String): Prison =
+    prisonRepository.findByGpPracticeGpPracticeCode(gpPracticeCode)
+      ?: throw EntityNotFoundException("Prison with gp practice $gpPracticeCode not found")
```
Comment (mikehalmamoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonregister/model/PrisonService.kt:24]: I couldn't find a test for the `throw` here - might have missed it though :smile:

**Review by mikehalmamoj — APPROVED** [review]

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonregister/model/PrisonService.kt (hunk 3 above)
Comment (petergphillips) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/prisonregister/model/PrisonService.kt:24]: will fix in next PR - I copied verbatim from prison estate so entirely likely missed a test along the way

**Pull request merged**